docs/getting-started-guides/mesos-docker.md

```diff
@@ -154,22 +161,38 @@ host machine (mac).
 
 1. Build binaries
 
+    You will need the server binaries compiled for linux/amd64 and kubectl compiled for your local architecture and operating system.
+
+    If you have Golang installed, you can build Kubernetes for your local architecture and operating system:
+
     ```
     KUBERNETES_CONTRIB=mesos hack/build-go.sh
     ```
 
-    Alternatively, you can use `make`, if make is installed.
+    If you have Golang and make installed, you can build Kubernetes for your local architecture and operating system with make:
+
+    ```
+    KUBERNETES_CONTRIB=mesos make
+    ```
 
-    Unless you're on linux, you'll also need to build the binaries for linux/amd64 (for the docker containers):
+    If you don't have or want Golang installed, you can build Kubernetes for linux/amd64 in a docker container:
```

Comment:
Do we really need this many cases?

Why not just
```
KUBERNETES_CONTRIB=mesos
make
build/run.sh hack/build-go.sh (optional for linux)
```
Full stop. For more information /docs/development.md.